### PR TITLE
Add Hostname helper

### PR DESCRIPTION
**Pull request by caseydavenport**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: lib/names/hostname.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package names
+
+import (
+	"os"
+	"strings"
+)
+
+// Hostname returns this hosts hostname, converting to lowercase so that
```
Comment (robbrockbank) [inline lib/names/hostname.go]: I think the IPAM package should be update to use this too.  Although one niggle with that is that we need IPAM to work on v2 and v3 simultaneously (in order to implement the proposed CNI changes)

File: lib/names/hostname.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// it is valid for use in the Calico API.
+func Hostname() (string, error) {
+	if h, err = os.Hostname(); err != nil {
+		return "", err
+	}
+	return strings.ToLower(strings.TrimSpace(h)), nil
```
Comment (robbrockbank) [inline lib/names/hostname.go]: either need this in an else or you need to pull the assignment out of the if statement.

File: lib/names/hostname.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import (
+	"os"
+	"strings"
+)
+
+// Hostname returns this hosts hostname, converting to lowercase so that
+// it is valid for use in the Calico API.
+func Hostname() (string, error) {
+	if h, err = os.Hostname(); err != nil {
```
Comment (robbrockbank) [inline lib/names/hostname.go]: = -> :=

File: lib/names/hostname.go (hunk 2 above)
Comment (robbrockbank) [reply lib/names/hostname.go]: whilst this covers immediate use case I'd like to be paranoid and handle things like leading/terminating dashes or dots as part of the upgrade work.

File: lib/names/hostname.go (hunk 1 above)
Comment (caseydavenport) [reply lib/names/hostname.go]: suppose the v3 IPAM call should use this, v2 should continue as usual.

**Comment by caseydavenport** [comment]
@robbrockbank CI is passing now, good to go?

**Pull request merged**